src/depthai_camera.cpp

```diff
@@ -0,0 +1,261 @@
+    int height = _videoHeight;
+    int fps = _videoFps;
+    int bitrate = _videoBitrate;
+    std::string encoding = _videoH265 ? "H265" : "H264";
+
+    if(ValidateParameters(width, height, fps, bitrate, encoding))
+    {
+        _videoWidth = width;
+        _videoHeight = height;
+        _videoFps = fps;
+        _videoBitrate = bitrate;
+        _videoH265 = (encoding == "H265");
+        TryRestarting();
+    }
+}
+
+void DepthAICamera::TryRestarting()
+{
+    if(_thread_running )
+    {
+        _thread_running = false;
+    }
+    if(_processing_thread.joinable())
+    {
+        _processing_thread.join();
+    }
+
+    RCLCPP_INFO(this->get_logger(), "[%s]: (Re)Starting...", get_name());
+
+    _pipeline = std::make_shared<dai::Pipeline>();
+
+    auto monoLeft = _pipeline->create<dai::node::MonoCamera>();
+    auto monoRight = _pipeline->create<dai::node::MonoCamera>();
+    auto colorCamera = _pipeline->create<dai::node::ColorCamera>();
+    auto videoEncoder = _pipeline->create<dai::node::VideoEncoder>();
+
+    auto xoutLeft = _pipeline->create<dai::node::XLinkOut>();
+    auto xoutRight = _pipeline->create<dai::node::XLinkOut>();
+    auto xoutColor = _pipeline->create<dai::node::XLinkOut>();
+    auto xoutVideo = _pipeline->create<dai::node::XLinkOut>();
+
+    // Setup Grayscale Cameras
+    monoLeft->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
+    monoLeft->setBoardSocket(dai::CameraBoardSocket::LEFT);
+    monoRight->setResolution(dai::MonoCameraProperties::SensorResolution::THE_720_P);
+    monoRight->setBoardSocket(dai::CameraBoardSocket::RIGHT);
+
+    // Setup Color Camera
+    colorCamera->setBoardSocket(dai::CameraBoardSocket::RGB);
+    colorCamera->setVideoSize(_videoWidth, _videoHeight);
+    colorCamera->setResolution(dai::ColorCameraProperties::SensorResolution::THE_1080_P);
+    colorCamera->setPreviewSize(_videoWidth, _videoHeight);
+
+    Profile encoding = _videoH265 ? Profile::H265_MAIN : Profile::H264_MAIN;
+    videoEncoder->setDefaultProfilePreset(_videoWidth, _videoHeight, static_cast<float>(_videoFps), encoding);
+    videoEncoder->setBitrate(_videoBitrate);
+
+    colorCamera->video.link(videoEncoder->input);
+    videoEncoder->bitstream.link(xoutVideo->input);
+    monoLeft->out.link(xoutLeft->input);
+    monoRight->out.link(xoutRight->input);
+    colorCamera->preview.link(xoutColor->input);
+
+    xoutLeft->setStreamName("left");
+    xoutRight->setStreamName("right");
+    xoutColor->setStreamName("color");
+    xoutVideo->setStreamName("video");
+
+    try
+    {
+        _device = std::make_shared<dai::Device>(*_pipeline, true);
+    }
+    catch (const std::runtime_error& err)
+    {
+        RCLCPP_ERROR(get_logger(), "Cannot start DepthAI camera: " + std::string(err.what()));
+        _device.reset();
+        return;
+    }
+
+    _leftQueue = _device->getOutputQueue("left", 30, false);
+    _rightQueue = _device->getOutputQueue("right", 30, false);
```

Comment:
Do we need such high frame rate for these cameras?

Reply:
It's not a framerate, it's depth of queue

Reply:
ok.